packages/block-editor/src/components/inspector-controls-tabs/index.js

```diff
@@ -11,28 +11,45 @@ import SettingsTab from './settings-tab';
 import StylesTab from './styles-tab';
 import InspectorControls from '../inspector-controls';
 import useIsListViewTabDisabled from './use-is-list-view-tab-disabled';
+import { store as blockEditorStore } from '../../store';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 export default function InspectorControlsTabs( {
 	blockName,
 	clientId,
 	hasBlockStyles,
 	tabs,
 } ) {
+	const { selectedTab } = useSelect( ( select ) => {
+		const { __unstableGetInspectorControlsTabSelectedTab: getSelectedTab } =
+			select( blockEditorStore );
+
+		const _selectedTab = getSelectedTab();
+
+		return {
+			selectedTab: _selectedTab,
+		};
+	}, [] );
+
+	const { __unstableSetInspectorControlsTabSelectedTab: setSelectedTab } =
+		useDispatch( blockEditorStore );
+
 	// The tabs panel will mount before fills are rendered to the list view
 	// slot. This means the list view tab isn't initially included in the
 	// available tabs so the panel defaults selection to the styles tab
 	// which at the time is the first tab. This check allows blocks known to
 	// include the list view tab to set it as the tab selected by default.
 	const initialTabName = ! useIsListViewTabDisabled( blockName )
 		? TAB_LIST_VIEW.name
-		: undefined;
+		: selectedTab;
```

Comment:
Note to self: we must ensure the default return value of the reducer is also `undefined`. This will ensure the default behaviour still applies when the state is not set in the editor store.